Userland/Libraries/LibGfx/ImageFormats/WebPLoaderLossy.cpp

```diff
@@ -1221,6 +1219,9 @@ ErrorOr<void> decode_VP8_image_data(Gfx::Bitmap& bitmap, FrameHeader const& head
         }
     }
 
+    for (auto& decoder : streams)
+        TRY(decoder.finish_decode());
```

Comment:
why add calls to this in the webp path? the vp8 spec doesn't say anything about the content of padding afaict.

(also, seems unrelated to the rest of the commit)

Reply:
As noted in the definition of `finish_decode()`, the padding in our WebP test inputs also fill the remaining data in the range with zeroes, so this can be used to verify that the data is consumed correctly. That check is only enabled when VPX_DEBUG is enabled. Given that it seems consistent, I see no reason to separate this functionality off from the general BooleanDecoder implementation.

The function is also used in the second commit to check for deferred overread errors, but both parts of the function are of benefit.

Reply:
If desired, the padding check can be made into a debug print instead of an error, though.

Reply:
Going to resolve this, but let me know if you would like the optional padding check to be a print instead of an error.